File: route/api_search.py
from .tool.func import *

def api_search(name = 'Test', num = 10, page = 1):
    with get_db_connect() as conn:
        curs = conn.cursor()

        num = 1 if num > 1000 else num
        page = (page * (num - 1)) if page * num > 0 else 0

        # 개편 예정
        curs.execute(db_change('select data from other where name = "count_all_title"'))
        if int(curs.fetchall()[0][0]) < 30000:
            curs.execute(db_change("" + \
                "select distinct title, case " + \
                "when title like ? then 'title' else 'data' end from data " + \
                "where (title like ? or data like ?) order by case " + \
                "when title like ? then 1 else 2 end limit ?, ?"),
                ['%' + name + '%', '%' + name + '%', '%' + name + '%', '%' + name + '%', page, num]
            )
        else:
            curs.execute(db_change("select title from data where title like ? order by title limit ?, ?"),
                ['%' + name + '%', page, num]
            )
            
        all_list = curs.fetchall()
        if all_list:
            return flask.jsonify(all_list)
        else:
            return flask.jsonify({})

File: route_go/route/api_search.go
package route

import (
	"encoding/json"
	"fmt"
	"log"
	"strconv"

	"opennamu/route/tool"
)

func Api_search(call_arg []string) {
	db_set := map[string]string{}
	json.Unmarshal([]byte(call_arg[0]), &db_set)

	other_set := map[string]string{}
	json.Unmarshal([]byte(call_arg[1]), &other_set)

	page, _ := strconv.Atoi(other_set["num"])
	num := 0
	if page*50 > 0 {
		num = page*50 - 50
	}

	db := tool.DB_connect(db_set)
	if db == nil {
		return
	}
	defer db.Close()

	if other_set["search_type"] == "title" {
		stmt, err := db.Prepare(tool.DB_change(db_set, "select title from data where title collate nocase like ? order by title limit ?, 50"))
		if err != nil {
			log.Fatal(err)
		}
		defer stmt.Close()

		var title string
		var title_list []string

		rows, err := stmt.Query("%"+other_set["name"]+"%", num)
		if err != nil {
			log.Fatal(err)
		}
		defer rows.Close()

		for rows.Next() {
			err := rows.Scan(&title)
			if err != nil {
				log.Fatal(err)
			}

			title_list = append(title_list, title)
		}

		if len(title_list) == 0 {
			fmt.Print("{}")
		} else {
			json_data, _ := json.Marshal(title_list)
			fmt.Print(string(json_data))
		}
	} else {
		stmt, err := db.Prepare(tool.DB_change(db_set, "select title from data where data collate nocase like ? order by title limit ?, 50"))
		if err != nil {
			log.Fatal(err)
		}
		defer stmt.Close()

		var title string
		var title_list []string

		rows, err := stmt.Query("%"+other_set["name"]+"%", num)
		if err != nil {
			log.Fatal(err)
		}
		defer rows.Close()

		for rows.Next() {
			err := rows.Scan(&title)
			if err != nil {
				log.Fatal(err)
			}

			title_list = append(title_list, title)
		}

		if len(title_list) == 0 {
			fmt.Print("{}")
		} else {
			json_data, _ := json.Marshal(title_list)
			fmt.Print(string(json_data))
		}
	}
}
